tempodb/backend/gcs/gcs.go

```diff
@@ -167,10 +175,14 @@ func (rw *readerWriter) ReadRange(ctx context.Context, name string, keypath back
 func (rw *readerWriter) Shutdown() {
 }
 
-func (rw *readerWriter) writer(ctx context.Context, name string) *storage.Writer {
-	w := rw.bucket.Object(name).NewWriter(ctx)
+func (rw *readerWriter) writer(ctx context.Context, name string) (*storage.Writer, error) {
```

Comment:
why did we add an error return here? you don't seem to be using it and it only complicates code above.

Reply:
There was an update call in a previous commit that returned an error.  If this doesn't end up returning an error, I agree we can drop the signature change.

Reply:
The `err` is back now with the use of `mergo`.

Reply:
i'll admit i really don't like the use of mergo here. likely everything would be fine, but this could also be a source of future head scratching bugs. I'd prefer either 

- manually copying over a set of static fields we support
OR
- only setting the ObjectAttrs.Metadata map. It's possible that setting a "Cache-Control" = "no-cache" in this map is translated by the API into the fixed attribute.

Reply:
If the ObjectAttrs.Metadata map is enough, that seems fine just to allow a `map[string]string` to be set in our config and then only update that.  I'm not sure without testing, but I wonder why there would be two methods to update the same value control.  Perhaps one is read only?  Reading here was what clued me in to where the attributes need to be set.

https://github.com/googleapis/google-cloud-go/blob/main/storage/storage.go#L1126

But I also see that the Metadata documentation mentions "user provided" fields.

Reply:
This doc makes me think `metadata` is only used for custom key-value pairs.

https://cloud.google.com/storage/docs/viewing-editing-metadata

The use of "standard" vs "custom" in the docs seems to indicate that to set a standard field like Cache-Control, you use the top level field, and for custom bits you use the metadata field.

I'll work up a test to see what it looks like.

Reply:
Setting the metadata doesn't appear to update the `CacheControl` field on the object.

https://gist.github.com/zalegrala/9d86cbeefa1ebb43bee90afd6e6d4600

The first test file was created with the `metadata` field set, and the second was created with the `CacheControl` field on the object set.

```
❯ gsutil ls -L gs://zalegrala-test/test1
gs://zalegrala-test/test1:
    Creation time:          Mon, 11 Apr 2022 20:43:16 GMT
    Update time:            Mon, 11 Apr 2022 20:43:16 GMT
    Storage class:          STANDARD
    Content-Length:         13
    Content-Type:           text/plain; charset=utf-8
    Metadata:               
        Cache-Control:      no-cache
    Hash (crc32c):          sqwNTw==
    Hash (md5):             KVH1pxihX0ylP/tHoGXnAw==
    ETag:                   COSVp/PvjPcCEAE=
    Generation:             1649709796674276
    Metageneration:         1
    ACL:                    []
TOTAL: 1 objects, 13 bytes (13 B)
```
```
❯ gsutil ls -L gs://zalegrala-test/test2
gs://zalegrala-test/test2:
    Creation time:          Mon, 11 Apr 2022 20:44:59 GMT
    Update time:            Mon, 11 Apr 2022 20:44:59 GMT
    Storage class:          STANDARD
    Cache-Control:          no-cache
    Content-Length:         13
    Content-Type:           text/plain; charset=utf-8
    Hash (crc32c):          sqwNTw==
    Hash (md5):             KVH1pxihX0ylP/tHoGXnAw==
    ETag:                   CMXJsaTwjPcCEAE=
    Generation:             1649709899605189
    Metageneration:         1
    ACL:                    []
TOTAL: 1 objects, 13 bytes (13 B)
```

Note the output only contains the cache control on the second.

Since it sounds like we are leaning to avoiding `mergo` and bringing in all possible attributes, which ones do we want to focus on supporting?  The issue we're aiming to fix here only mentions cache-control, but perhaps there are other common uses.

Reply:
Hmm, `CacheControl` and `Metadata` ? Do any other fields make sense to you?

Reply:
Those sound like a good start to me.  We can increment if someone else wants something later.